src/Compilers/CSharp/Portable/Symbols/Source/SourceComplexParameterSymbol.cs

```diff
@@ -248,23 +248,23 @@ private Binder GetBinder(SyntaxNode syntax)
 
         private void NullableAnalyzeParameterDefaultValueFromAttributes()
         {
-            if (!NullableWalker.NeedsAnalysis(DeclaringCompilation))
+            var parameterSyntax = this.CSharpSyntaxNode;
+            if (parameterSyntax == null)
             {
+                // If there is no syntax at all for the parameter, it means we are in a situation like
+                // a property setter whose 'value' parameter has a default value from attributes.
+                // There isn't a sensible use for this in the language, so we just bail in such scenarios.
                 return;
             }
 
-            var defaultValue = DefaultValueFromAttributes;
-            if (defaultValue == null || defaultValue.IsBad)
+            if (!NullableWalker.NeedsAnalysis(DeclaringCompilation, parameterSyntax))
```

Comment:
>parameterSyntax [](start = 68, length = 15)

Should we take value's context into account?

Reply:
> Should we take value's context into account?

Is the suggestion to include the nullable context of the `[DefaultParameterValue(value)]` attribute? If so, that should be part of `parameterSyntax`.

---
In reply to: [544687774](https://github.com/dotnet/roslyn/pull/49953#discussion_r544687774) [](ancestors = 544687774)

Reply:
Are there any tests with the `#nullable enable` inside the attribute itself? Such as:
```cs
[DefaultValue(
#nullable enable
null
#nullable disable
)]
```

---
In reply to: [545604160](https://github.com/dotnet/roslyn/pull/49953#discussion_r545604160) [](ancestors = 545604160,544687774)